src/LoginForm/LoginForm.test.js

```diff
@@ -41,4 +41,14 @@ describe("LoginForm", () => {
     fireEvent.submit(elements.submit);
     await waitFor(() => expect(onLogin).not.toHaveBeenCalled());
   });
+  it("displays error message to user on validation fail", async () => {
+    const elements = setup();
+    userEvent.type(elements.inputs.email, "joe.bloggs"); // missing password
```

Comment:
shouldn't userEvent be wrapped in act @mattcorner ?

Reply:
You'd think so, but not in this case because the form validation is only applied onSubmit, not onTouch like it is in something like RegistrationForm.

Reply:
Also, it's testing wrong email not missing password as the comment suggests. Password is included on this form but that's not what you are testing for.

Reply:
> You'd think so, but not in this case because the form validation is only applied onSubmit, not onTouch like it is in something like RegistrationForm.

So we're saying the rule is only wrap userEvent with act if the userEvent call triggers some async code from the react-hook-form library which updates the DOM? In all other cases you don't need to wrap userEvent?